src/engagement_db_to_analysis/membership_group.py

```diff
@@ -0,0 +1,90 @@
+import os
+from google.api_core.exceptions import  NotFound
+import csv
+
+from core_data_modules.logging import Logger
+from core_data_modules.util import IOUtils
+from storage.google_cloud import google_cloud_utils
+from core_data_modules.util import TimeUtils
+from core_data_modules.traced_data import Metadata
+
+log = Logger(__name__)
+
+
+def get_membership_groups_csvs(google_cloud_credentials_file_path, membership_group_csv_urls, membership_group_dir_path):
+    """
+    Downloads de-identified membership groups CSVs from g-cloud.
+    :param google_cloud_credentials_file_path: Path to the Google Cloud service account credentials file to use to
+                                               access the credentials bucket.
+    :type google_cloud_credentials_file_path: str
+    :param membership_group_csv_urls: Dict of membership group name to group g-cloud csv url(s).
+    :type membership_group_csv_urls: Dict
+    :param membership_group_dir_path: Path to directory containing de-identified membership groups CSVs containing membership groups data
+                        stored as `avf-participant-uuid` column.
+    """
+
+    for membership_group, membership_group_csv_url in membership_group_csv_urls:
+        for i, membership_group_csv_url in enumerate(membership_group_csv_url):
+            membership_group_csv = membership_group_csv_url.split("/")[-1]
+
+            export_file_path = f'{membership_group_dir_path}/{membership_group_csv}'
+
+            if os.path.exists(export_file_path):
+                log.info(f"File '{membership_group_csv}' already exists, skipping download")
+                continue
+
+            try:
+                log.info(f"Saving '{membership_group_csv}' to file f'{membership_group_dir_path}...")
+                IOUtils.ensure_dirs_exist_for_file(export_file_path)
+                with open(export_file_path, "wb") as membership_group_csv_file:
+                    google_cloud_utils.download_blob_to_file(
+                        google_cloud_credentials_file_path, membership_group_csv_url, membership_group_csv_file)
+
+            except NotFound:
+                log.warning(f"{membership_group_csv}' not found in google cloud, skipping download")
+
+
+def tag_membership_groups_participants(user, column_view_traced_data, membership_group_csv_urls, membership_group_dir_path):
+    """
+    This tags uids who participated in projects membership groups.
+    :param user: Identifier of the user running this program, for TracedData Metadata.
+    :type user: str
+    :param column_view_traced_data: Messages/Participants TracedData organised into column-view format.
+    :type: list of core_data_modules.traced_data.TracedData
+    :param membership_group_dir_path: Path to directory containing de-identified membership groups CSVs containing membership groups data
+                        stored as `avf-phone-uuid` and `Name` columns.
+    :type user: str
```

Comment:
replace `:type user: str` with `:type membership_group_dir_path: str`

Reply:
Thanks for catching this - updated.